app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java

```diff
@@ -3,16 +3,18 @@
 import android.os.Parcel;
 import android.os.Parcelable;
 
-import com.j256.ormlite.stmt.SelectArg;
 import com.j256.ormlite.stmt.Where;
 import com.zulip.android.models.Message;
 import com.zulip.android.models.Stream;
+import com.zulip.android.util.Constants;
 
 import org.json.JSONException;
 
 import java.sql.SQLException;
 import java.text.SimpleDateFormat;
+import java.util.Calendar;
 import java.util.Date;
+import java.util.Locale;
 
 public class NarrowFilterByDate implements NarrowFilter {
```

Comment:
Also I just noticed, that we are not using writeToParcel method in NarrowFilterByDate. When we need to pass objects across activities we implement the [Parcelable interface](https://developer.android.com/reference/android/os/Parcelable.html) for those object classes. We are saving NarrowFilter instances in a [bundle](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/MessageListFragment.java#L81) to save the state. See [NarrowFilterStream](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterStream.java) for example.